src/shared/services/auth/auth.js

```diff
@@ -113,6 +103,50 @@ function handleCallback() {
   global.history.replaceState('', document.title, global.location.pathname);
 }
 
+/**
+ * Fetch token with given authorization code.
+ */
+async function fetchTokenByCode(code) {
+  const { authEndpoint, realm, clientId } = configuration.keycloak;
+
+  const searchParams = new URLSearchParams({
+    code,
+    grant_type: 'authorization_code',
+    client_id: clientId,
+    redirect_uri: AUTH_REDIRECT_URI,
+  });
+
+  const tokenUrl = `${authEndpoint}/realms/${realm}/protocol/openid-connect/token`;
+
+  const response = await fetch(tokenUrl, {
+    body: searchParams.toString(),
+    method: 'POST',
+    headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+  });
+
+  return response.json();
+}
+
+/**
+ * Handle login callback.
+ */
+async function handleCallback() {
+  // Parse query string into object
+  const params = queryStringParser(global.location.hash);
+
+  if (!params || !params.state) return;
+
+  if (params.code) { // handling keycloak callback
```

Comment:
Maybe we can here in the comments make distinction between code and token flow and not specific keycloak an other callbacks.